packages/@webex/plugin-meetings/src/recording-controller/index.ts

```diff
@@ -276,14 +278,25 @@ export default class RecordingController {
    * @returns {Promise}
    */
   private recordingFacade(action: RecordingAction): Promise<any> {
+    const isPremiseRecordingEnabled = Util.isPremiseRecordingEnabled(
+      this.displayHints,
+      this.selfUserPolicies
+    );
     LoggerProxy.logger.log(
       `RecordingController:index#recordingFacade --> recording action [${action}]`
     );
 
+    let recordingType: RecordingType;
+    if (isPremiseRecordingEnabled) {
```

Comment:
is this truly an if else? if it's premise based it's never an option not to also have cloud ?

Reply:
Yes, we can have either cloud OR premise recording. Both cloud and premise recording cannot be available together.